src/normalise/v3/normaliseSecurity.js

```diff
@@ -4,6 +4,7 @@ const normaliseV2Security = require('../v2/normaliseSecurity')
 const normaliseSecurity = security => {
   if (!security) return
   const [first] = security
+  if (!first) return null
```

Comment:
I get why you return `null` here, to distinguish between `undefined` security (in which case the default gets applied) vs there explicitly being no security. It feels like a security block without any actual elements is more of an error condition however and so this should throw an error which could be caught by `extractPaths` — that might make the logic a bit cleaner.

Open to your thoughts on this however.

Reply:
As stated in the documentation, when root level security is defined, you can override it by specifying operation level security. Either to use a different scheme, or to use **no** security.
> After you have defined the security schemes in the securitySchemes section, you can apply them to the whole API or individual operations...

And then later on:
>Global security can be overridden in individual operations to use a different authentication type, different OAuth/OpenID scopes, or no authentication at all...

You can read the whole section: https://swagger.io/docs/specification/authentication/#security

Maybe an error should be thrown, when root level security is not defined?

Reply:
Rot level security is not _required_ however so not defining one is not strictly an error.  I am going to approve this for now but maybe have some thought about it in future.